src/Form/AdminSettingsForm.php

```diff
@@ -0,0 +1,399 @@
+      '#default_value' => $visibility_settings['path_patterns'],
+    ];
+
+    return $form;
+  }
+
+  /**
+   * Display thumbnail URL form.
+   *
+   * @return array
+   *   Thumbnail URL form.
+   */
+  private function buildThumbnailUrlForm() {
+    $form = [
+      '#title' => t('Thumbnail URL'),
+      '#type' => 'details',
+      '#tree' => TRUE,
+      '#group' => 'data_collection_settings',
+    ];
+    $form['link_list'] = [
+      '#type' => 'markup',
+      '#markup' => '<div>' . t('There are no content types. Please create a content type first.') . '</div>',
+    ];
+
+    $node_types = NodeType::loadMultiple();
+    if (empty($node_types)) {
+      return $form;
+    }
+
+    $links = [];
+    $link_attributes = ['attributes' => ['target' => '_blank'], 'fragment' => 'edit-acquia-lift'];
+    foreach ($node_types as $node_type) {
+      $url = Url::fromRoute('entity.node_type.edit_form', ['node_type' => $node_type->id()], $link_attributes);
+      $links[] = '<p>' . \Drupal::l($node_type->label(), $url) . '</p>';
+    }
+    $form['link_list']['#markup'] = t('Configure thumbnail URLs on each content type\'s edit page (in a new window):');
+    $form['link_list']['#markup'] .= implode('', $links);
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $settings = $this->config('acquia_lift.settings');
+    $values = $form_state->getValues();
+
+    $this->setCredentialValues($settings, $values['credential']);
+    $this->setIdentityValues($settings, $values['identity']);
+    $this->setFieldMappingsValues($settings, $values['field_mappings']);
+    $this->setVisibilityValues($settings, $values['visibility']);
+
+    $settings->save();
+
+    drupal_flush_all_caches();
+
+    parent::submitForm($form, $form_state);
+  }
+
+  /**
+   * Set credential values.
+   *
+   * @param \Drupal\Core\Config\Config $settings
+   *   Acquia Lift config settings.
+   * @param array $values
+   *   Credential values.
+   */
+  private function setCredentialValues(Config $settings, array $values) {
+    $settings->set('credential.account_name', $values['account_name']);
+    $settings->set('credential.customer_site', $values['customer_site']);
+    $settings->set('credential.api_url', $this->removeProtocal($values['api_url']));
+    $settings->set('credential.access_key', $values['access_key']);
+    if (!empty($values['secret_key'])) {
+      $settings->set('credential.secret_key', $values['secret_key']);
+    }
+    $settings->set('credential.js_path', $this->removeProtocal($values['js_path']));
+  }
+
+  /**
+   * Remove the protocal from an URL string.
+   *
+   * @param string $url
+   *   URL.
+   *
+   * @return string
+   *   Same URL as input except with the 'http://' and 'https://' trimmed.
+   */
+  private function removeProtocal($url) {
+    return preg_replace('~^https?://~', '', $url);
```

Comment:
I didn't have to because I used "~" as the delimiter instead of the usual "/". I tested this again and it appears to work well.